sabnzbd/utils/correct_extension.py

```diff
@@ -0,0 +1,316 @@
+    "odt",
+    "oft",
+    "ogg",
+    "ost",
+    "otf",
+    "part",
+    "pdf",
+    "php",
+    "php",
+    "pkg",
+    "pl",
+    "pl",
+    "pl",
+    "png",
+    "pps",
+    "ppt",
+    "pptx",
+    "ps",
+    "psd",
+    "pst",
+    "py",
+    "py",
+    "py",
+    "rar",
+    "rm",
+    "rpm",
+    "rss",
+    "rtf",
+    "sav",
+    "sh",
+    "sql",
+    "svg",
+    "swf",
+    "swift",
+    "sys",
+    "tar",
+    "tar",
+    "gz",
+    "tex",
+    "tif",
+    "tiff",
+    "tmp",
+    "toast",
+    "ttf",
+    "txt",
+    "vb",
+    "vcd",
+    "vcf",
+    "vob",
+    "wav",
+    "wma",
+    "wmv",
+    "wpd",
+    "wpl",
+    "wsf",
+    "xhtml",
+    "xls",
+    "xlsm",
+    "xlsx",
+    "xml",
+    "z",
+    "zip",
+)
+
+DOWNLOAD_EXT = (
+    "ass",
+    "avi",
+    "bat",
+    "bdmv",
+    "bin",
+    "bup",
+    "clpi",
+    "crx",
+    "db",
+    "diz",
+    "docx",
+    "epub",
+    "exe",
+    "flac",
+    "gif",
+    "gz",
+    "htm",
+    "html",
+    "icns",
+    "ico",
+    "idx",
+    "ifo",
+    "img",
+    "inf",
+    "info",
+    "ini",
+    "iso",
+    "jpg",
+    "log",
+    "m2ts",
+    "m3u",
+    "m4a",
+    "mkv",
+    "mp3",
+    "mp4",
+    "mpls",
+    "mx",
+    "nfo",
+    "nib",
+    "nzb",
+    "otf",
+    "par2",
+    "part",
+    "pdf",
+    "pem",
+    "php",
+    "plist",
+    "png",
+    "py",
+    "rar",
+    "releaseinfo",
+    "rev",
+    "sfv",
+    "sh",
+    "srr",
+    "srs",
+    "srt",
+    "strings",
+    "sub",
+    "sup",
+    "sys",
+    "tif",
+    "ttf",
+    "txt",
+    "url",
+    "vob",
+    "website",
+    "wmv",
+    "xml",
+    "xpi",
+)
+
+RAR_EXT = tuple(["r" + str(i).zfill(2) for i in range(0, 100)])  # "r00", "r01", ...
+
+# combine to one tuple, with unique entries:
+ALL_EXT = tuple(set(COMMON_EXT + DOWNLOAD_EXT + RAR_EXT))
+
+
+def has_common_extension(file_path: str) -> int:
+    """returns boolean if the extension of file_path is a common, well-known extension"""
+    # TBD use SAB's own extension finder
+    filename, file_extension = os.path.splitext(file_path)
+    file_extension = file_extension[1:].lower()
+    return file_extension in ALL_EXT
+
+
+def all_possible_extensions(file_path: str) -> list:
+    """returns a list with all possible extensions for given file_path"""
+    extension_list = []
+    for i in puremagic.magic_file(file_path):
+        extension_list.append(i.extension)
+    return extension_list
+
+
+def most_likely_extension(file_path: str) -> str:
+    """Returns most_likely extension"""
+    for possible_extension in all_possible_extensions(file_path):
+        # let's see if technically-suggested extension is also likely IRL
+        if possible_extension in ALL_EXT:
+            # Yes, looks likely
+            return possible_extension
+    # no common extension found, so just return the first
+    try:
+        return all_possible_extensions(file_path)[0]
+    except:
+        return ""
+
+
+def extension_matches(file_path: str) -> int:
```

Comment:
this function too returns a boolean. Also consider making the function name reflect that (e.g. _is_this()_, _has_that()_ or similar).

Reply:
Yes, I have to review the naming of the functions and the module itself.